mlrun/api/launcher.py

```diff
@@ -147,8 +148,14 @@ def verify_base_image(runtime):
         pass
 
     @staticmethod
-    def _enrich_runtime(runtime):
-        pass
+    def _enrich_runtime(runtime: "mlrun.runtimes.base.BaseRuntime"):
+        if runtime.metadata.project:
+            project = runtime._get_db().get_project(runtime.metadata.project)
+            # this is mainly for tests with nop db
+            # in normal use cases if no project is found we will get an error
+            if project:
+                project = mlrun.projects.project.MlrunProject.from_dict(project.dict())
+                project.enrich_function_object(runtime, copy_function=False)
```

Comment:
add comment why we enrich the function of object from the runtime only on server side